### ISSUE TITLE
Fix Missing Author IDs in Pull Requests and Issues Tables

### ISSUE THREAD

**Issue by babblebey** [bdougie/contributor.info#736]
We have a significant data integrity issue in our Supabase database where the `author_id` field is `null` in many rows of both the `pull_requests` and `issues` tables. This prevents proper relationship mapping to the `contributors` table and affects contributor analytics and insights.

### Sample Affected Row
```json
{
  "idx": 0,
  "id": "5e2f4537-f830-48ce-a284-c78cf792af12",
  "github_id": 506398625,
  "number": 46572,
  "title": "[pytorch] Early return in nn.EmbeddingBag when weight is empty",
  "body": "Summary: When `num_samples == 0`, grid becomes zero...",
  "state": "closed",
  "repository_id": "a87980e2-e273-4e87-925b-b72559994b2b",
  "author_id": null, // ← This should be a UUID from contributors table
  "assignee_id": null,
  "html_url": "https://github.com/pytorch/pytorch/pull/46572",
  // ... other fields
}
```

## Scope

- **Affected tables**: `pull_requests` and `issues`
- **Estimated affected rows**: 100,000+
- **Data source**: GitHub API data that was synced without proper author relationship mapping

## Proposed Solution

Create a JavaScript script that:

1. **Fetches missing author data** from GitHub API using the `html_url` field
2. **Parses GitHub URLs** to extract `owner/repo/number` for API calls
3. **Queries contributors table** to check if author already exists (by `github_id`)
4. **Creates new contributor records** if they don't exist
5. **Updates author_id fields** with the proper UUID references
6. **Processes in batches** to handle large volumes efficiently

#### Phase 1: Setup & Analysis
- [x] Analyze scope of missing data in both tables
- [x] Create replica tables for safe testing
- [ ] Set up GitHub API integration with proper rate limiting

#### Phase 2: Core Script Development
- [ ] URL parsing logic for GitHub pull requests and issues
- [ ] GitHub API client with retry logic and error handling
- [ ] Contributor lookup/creation functionality
- [ ] Batch processing system (100-500 rows per batch)

#### Phase 3: Execution & Monitoring
- [ ] Test on replica tables first
- [ ] Monitor GitHub API rate limits
- [ ] Process production data in off-peak hours
- [ ] Verify data integrity post-processing

### Technical Approach

```javascript
// Key components needed:
1. GitHub API integration (REST API for PR/issue details)
2. Supabase client for database operations  
3. Batch processing with configurable delays
4. Comprehensive error handling and retry logic
5. Progress tracking and logging
```

### Sample Implementation Structure

```javascript
class AuthorIdFixer {
  async fetchAuthorFromGithub(htmlUrl) {
    // Parse URL: https://github.com/pytorch/pytorch/pull/46572
    const { owner, repo, type, number } = this.parseGithubUrl(htmlUrl)
    
    // Use appropriate endpoint
    const endpoint = type === 'pull' 
      ? `repos/${owner}/${repo}/pulls/${number}`
      : `repos/${owner}/${repo}/issues/${number}`
      
    // Fetch with rate limiting and retry logic
    const response = await this.githubClient.get(endpoint)
    return response.data.user
  }
  
  async findOrCreateContributor(githubUser) {
    // Check if contributor exists by github_id
    // Create new contributor if needed
    // Return contributor UUID
  }
  
  async processBatch(tableName, rows) {
    // Process 100-500 rows at a time
    // Update author_id fields
    // Handle errors gracefully
  }
}
```

### API Endpoints Required

- `GET /repos/{owner}/{repo}/pulls/{number}` - For pull requests
- `GET /repos/{owner}/{repo}/issues/{number}` - For issues

### Environment Requirements

Using existing `.env.local` variables:
- `VITE_SUPABASE_URL` - Supabase project URL  
- `SUPABASE_SERVICE_ROLE_KEY` - Service role key for database operations
- `VITE_GITHUB_TOKEN` - GitHub API token for fetching author data

## Success Criteria

- [ ] All rows with valid `html_url` have corresponding `author_id`
- [ ] New contributors are properly created in the `contributors` table
- [ ] No data loss or corruption during the process
- [ ] Contributor analytics work correctly post-fix
- [ ] Performance impact is minimal during execution

## Additional Context

This issue directly impacts the contributor analytics and insights features of the platform. Missing author relationships prevent accurate:
- Contributor ranking calculations
- Pull request attribution  
- Organization-level contributor tracking
- Monthly leaderboard generation

### Example Script Usage
```bash
# Install dependencies
npm install @supabase/supabase-js node-fetch

# Set environment variables in .env.local
VITE_SUPABASE_URL=https://********.supabase.co
SUPABASE_SERVICE_ROLE_KEY=***eyJhbGc***
VITE_GITHUB_TOKEN=****

# Run the fix script
node fix-author-ids.js
```

**Comment by babblebey** [comment]
> * [x]  Analyze scope of missing data in both tables[x]  Create replica tables for safe testing
> * [x]  Create replica tables for safe testing

Created Replica Table for the following

- `contributors` - `contributors_replica`
- `pull_requests` - `pull_requests_replica` (copied 1000 rows for initial testing)
- `issues` - `issues_replica` (copied 1000 rows for initial testing)

**Comment by bdougie** [comment]
Does this work unblock this PR? #706

**Comment by babblebey** [comment]
Yea, It will address the `unknown` display name/usernames.

**Comment by babblebey** [comment]
I will now run the script against the production tables starting with only the `issues` for now.... I have taken a backup of each tables...

- `contributors` - `contributors_backup`
- `pull_requests` - `pull_requests_backup` 
- `issues` - `issues_backup` 

Its honestly scary to be doing this now, but I have logically classed this as a low risk operation, considering

- `contributors` table only gets a SELECT query for existing users and INSERT for new record, posing no risk to existing data
- `pull_requests` table just gets a SELECT query to find the faulty fields and UPDATE them
- `issuee` table just get a SELECT query to find the faulty fields and UPDATE them

### 🤞🏾

**Comment by babblebey** [comment]
Scripts running.. but no change is happening on the database.... Its RLS Security.... Can we temporarily disable that??? I've gotta run now... I'll let you know when's a good time to do, - if we can
